app/backend/src/couchers/servicers/communities.py

```diff
@@ -244,26 +244,25 @@ def RemoveAdmin(self, request, context, session):
 
     def ListMembers(self, request, context, session):
         page_size = min(MAX_PAGINATION_LENGTH, request.page_size or MAX_PAGINATION_LENGTH)
-        next_member_id = int(request.page_token) if request.page_token else 0
+        next_member_id = int(request.page_token) if request.page_token else None
+
         node = session.execute(select(Node).where(Node.id == request.community_id)).scalar_one_or_none()
         if not node:
             context.abort(grpc.StatusCode.NOT_FOUND, errors.COMMUNITY_NOT_FOUND)
-        members = (
-            session.execute(
-                select(User)
-                .join(ClusterSubscription, ClusterSubscription.user_id == User.id)
-                .where_users_visible(context)
-                .where(ClusterSubscription.cluster_id == node.official_cluster.id)
-                .where(User.id >= next_member_id)
-                .order_by(User.id)
-                .limit(page_size + 1)
-            )
-            .scalars()
-            .all()
+
+        query = (
+            select(User)
+            .join(ClusterSubscription, ClusterSubscription.user_id == User.id)
+            .where_users_visible(context)
+            .where(ClusterSubscription.cluster_id == node.official_cluster.id)
         )
+        if next_member_id is not None:
+            query = query.where(User.id < next_member_id)
+        members = session.execute(query.order_by(User.id.desc()).limit(page_size + 1)).scalars().all()
+
         return communities_pb2.ListMembersRes(
             member_user_ids=[member.id for member in members[:page_size]],
-            next_page_token=str(members[-1].id) if len(members) > page_size else None,
+            next_page_token=str(members[page_size - 1].id) if len(members) > page_size - 1 else None,
```

Comment:
The original here still works. This will now give an incorrect answer though if the length of the list is exactly the number of items left (i.e. you list 10 members out of 10, it will return a next page token even if there is no next page)